devel/lazarus/Portfile

```diff
@@ -0,0 +1,131 @@
+                    compatible with Delphi's Visual Component Library \
+                    (VCL).  The Lazarus Component Library (LCL) includes \
+                    equivalents for many of the familiar VCL controls such \
+                    as forms, buttons, text boxes and so on, which are used \
+                    to create applications with a graphical user interface \
+                    (GUI).
+
+homepage            https://wiki.freepascal.org/Main_Page
+master_sites        sourceforge:lazarus
+set src             ${name}-${version}.tar.gz
+
+checksums           rmd160 72cf729452ef651418d8b5fae8a1094ae9823212 \
+                    sha256 e01d10c1b08fdb9b98fa8f740536add804383e4f39f092059c5e4ac7516c147b \
+                    size   65556293
+
+depends_lib         port:fpc port:fpc-sources
+supported_archs     x86_64
+installs_libs       yes
+universal_variant   no
+
+extract.only        ${src}
+
+use_configure       no
+worksrcdir          lazarus
+
+post-patch {
+# add the MacPorts directory tree. This might need a check in a new version
+    reinplace "s|1\.\.15|1\.\.16|g"                                                       ${worksrcpath}/ide/include/unix/lazbaseconf.inc
+    reinplace "s|'/usr/share/fpcsrc'|'/usr/share/fpcsrc','${prefix}/share/fpcsrc'|g"      ${worksrcpath}/ide/include/unix/lazbaseconf.inc
+    reinplace "s|1\.\.8|1\.\.9|g"                                                         ${worksrcpath}/ide/include/unix/lazbaseconf.inc
+    reinplace "s|'/usr/share/lazarus',|'/usr/share/lazarus','${prefix}/share/lazarus',|g" ${worksrcpath}/ide/include/unix/lazbaseconf.inc
+
+# change the preferences directory
+#   reinplace "s|'\.lazarus'|'Library/Preferences/lazarus'|g"                              ${worksrcpath}/ide/include/unix/lazbaseconf.inc
+    reinplace "s|'/etc/lazarus'|'${prefix}/etc/lazarus'|g"                                 ${worksrcpath}/ide/include/unix/lazbaseconf.inc
+
+    reinplace "s|AddFilenameToList(List,'/usr/local/bin/'+GetDefaultCompilerFilename);|AddFilenameToList(List,'${prefix}/bin/'+GetDefaultCompilerFilename); AddFilenameToList(List,'/usr/local/bin/'+GetDefaultCompilerFilename);|g" ${worksrcpath}/ide/include/unix/lazbaseconf.inc
+
+# adjust the default paths to MacPorts paths
+    reinplace "s|/usr/local/bin/_PPCARCH_|${prefix}/bin/ppcx64|g" ${worksrcpath}/tools/install/macosx/environmentoptions.xml
+    reinplace "s|/usr/local/share|${prefix}/share|g"              ${worksrcpath}/tools/install/macosx/environmentoptions.xml
+    reinplace "s|/Developer/lazarus|${prefix}/share/lazarus|g"    ${worksrcpath}/tools/install/macosx/environmentoptions.xml
+
+    reinplace "s|<DebuggerFilename Value=\"/usr/bin/lldb\"/>|<DebuggerFilename Value=\"/usr/bin/lldb\"> <History Count=\"3\"> <Item1 Value=\"/usr/bin/lldb\"/> <Item2 Value=\"/usr/bin/gdb\"/> <Item3 Value=\"${prefix}/bin/gdb\"/> </History> </DebuggerFilename>|g" ${worksrcpath}/tools/install/macosx/environmentoptions.xml
```

Comment:
Would a patch be more readable than this? (I'm really asking, not telling.)

Reply:
I was also back and forth with this. Having it here at one place has its advantage and makes changes with a new version easier. I still might reconsider this.

Reply:
commit #faa51bd is my suggestion to make the patches a bit nicer.